src/buffer/out/Row.cpp

```diff
@@ -1181,3 +1181,36 @@ CharToColumnMapper ROW::_createCharToColumnMapper(ptrdiff_t offset) const noexce
     const auto guessedColumn = gsl::narrow_cast<til::CoordType>(clamp(offset, 0, _columnCount));
     return CharToColumnMapper{ _chars.data(), _charOffsets.data(), lastChar, guessedColumn };
 }
+
+const std::optional<ScrollbarData>& ROW::GetScrollbarData() const noexcept
+{
+    return _promptData;
+}
+void ROW::SetScrollbarData(std::optional<ScrollbarData> data) noexcept
+{
+    _promptData = data;

```

Comment:
As the `_promptData` doesn't get reset in `Reset()` which is called during buffer circling. Due to this I believe this code will start failing to work once the buffer is full.

If I'm correct, then simply adding it to `Reset()` should fix it.

Reply:
(This is my only remaining important concern. The memory layout is IMO not optimal for perf, but we can noodle on that later.)